src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs

```diff
@@ -308,5 +308,106 @@ private static IEnumerable<AnalyzerPerformanceInfo> Convert(IEnumerable<(Diagnos
         {
             return analyzerPerf.Select(kv => new AnalyzerPerformanceInfo(kv.analyzer.GetAnalyzerId(), DiagnosticAnalyzerLogger.AllowsTelemetry(kv.analyzer, serviceOpt), kv.timeSpan));
         }
+
+        public static bool IsCompilationEndAnalyzer(this DiagnosticAnalyzer analyzer, Project project, Compilation compilation)
+        {
+            // currently, this is only way to see whether analyzer has compilation end analysis or not.
```

Comment:
If we already have the compilation, then why not create a new `CompilationWithAnalyzers` instance with the analyzer and then invoke [GetAnalyzerTelemetryInfoAsync](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/CompilationWithAnalyzers.cs,1153)? This will just force complete all the action registration, and the returned info has the action counts.

Reply:
looked that API/implementation. it was doing way more work than the simple answer I wanted (it shows by the fact that it is Async method even though CompilationWithAnalyzer already has compilation in it). so, went this pass. do you think it will miss some cases?

example of more works, say analyzer register other hooks, call RunConcurrent or other APIs under initialize, CompilationWithAnalyzer actually do real work for all those calls, which I don't care at all for this question (is compilation end analzyer?), so basically the code does the minimum work to answer the question.